kernel/src/main/scala/cats/kernel/std/map.scala

```diff
@@ -1,16 +1,11 @@
 package cats.kernel
 package std
 
-import cats.kernel.std.util.StaticMethods.{ addMap, subtractMap }
+import cats.kernel.std.util.StaticMethods.addMap
 
 package object map extends MapInstances
 
-trait MapInstances extends MapInstances0 {
-  implicit def mapGroup[K, V: Group]: MapGroup[K, V] =
-    new MapGroup[K, V]
-}
-
-trait MapInstances0 {
+trait MapInstances {
```

Comment:
below in `MapEq` should we add the optimization of `(x eq y) || { // rest of the method?` no need to do `O(N)` work when they are referentially equal, right?